### PR TITLE
Add unit tests for ads history

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/13257

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [x] New files have MPL-2.0 license header.
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_ads/browser/ads_service_impl.h [hunk 1]
```diff
@@ -88,8 +88,6 @@ class AdsServiceImpl : public AdsService,
   void SetAdsPerHour(
       const uint64_t ads_per_hour) override;
 
-  uint64_t GetAdsPerDay() const override;
```
Comment (moritzhaller) [inline components/brave_ads/browser/ads_service_impl.h:91]: I did a quick search in b-c, is it correct that `GetAdsPerDay` was never used? How did we enforce  ads per day before if `GetAdsPerDay` was never called?

Comment (tmancey) [reply components/brave_ads/browser/ads_service_impl.h:91]: This was pref was used in the frequency cap. However, there was no need to expose this value via a pref as the value was never exposed in the UI. In the Muon days you could set ads per hour and ads per day in the UI.

**Comment by tmancey** [comment]
CI failed for unrelated reasons

**Review by kylehickinson — APPROVED** [review]
iOS lgtm

File: vendor/bat-native-ads/src/bat/ads/internal/ads_history/ads_history_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  history::AddNewTabPageAd(ad, ConfirmationType::kViewed);
+  history::AddNewTabPageAd(ad, ConfirmationType::kClicked);
+
+  // Assert
+  const std::deque<AdHistoryInfo> history = Client::Get()->GetAdsHistory();
+  ASSERT_EQ(2UL, history.size());
+}
+
+TEST_F(BatAdsAdsHistoryTest,
+    NewTabPageAdRespectsMaximumHistorySize) {
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/ads_history/ads_history_unittest.cc:96]: nit: maybe naming same as above

**Review by moritzhaller — APPROVED** [review]
LGTM++

File: vendor/bat-native-ads/src/bat/ads/internal/ads_history/ads_history_unittest.cc [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  history::AddAdNotification(ad, ConfirmationType::kViewed);
+  history::AddAdNotification(ad, ConfirmationType::kClicked);
+
+  // Assert
+  const std::deque<AdHistoryInfo> history = Client::Get()->GetAdsHistory();
+  ASSERT_EQ(2UL, history.size());
+}
+
+TEST_F(BatAdsAdsHistoryTest,
+    AdNotificationRespectsMaximumHistorySize) {
```
[the remark this answers is not in the document]
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads_history/ads_history_unittest.cc]: Fixed

File: vendor/bat-native-ads/src/bat/ads/internal/ads_history/ads_history_unittest.cc (hunk 2 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads_history/ads_history_unittest.cc]: Fixed

**Comment by tmancey** [comment]
CI failed for known reasons

**Pull request merged**